### PR TITLE
[bitnami/valkey-cluster] Update all the references from  to master/slave to primary/replica

### PR DESCRIPTION
**Pull request by dgomezleon**
### Description of the change

This major updates all the references from `master/slave` to `primary/replica` to follow the upstream project strategy. Environment variables previously prefixed as `VALKEY_MASTER` or `VALKEY_SENTINEL_MASTER` use `VALKEY_PRIMARY` and `VALKEY_SENTINEL_PRIMARY` now.

### Benefits

Use upstream terminology

### Possible drawbacks

Setting need to be updated to match `primary` instead of `master`

### Applicable issues

- fixes #

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/valkey-cluster/README.md [hunk 1]
```diff
@@ -25,13 +25,13 @@ Bitnami charts can be used with [Kubeapps](https://kubeapps.dev/) for deployment
...
+While [Valkey Helm Chart](https://github.com/bitnami/charts/tree/main/bitnami/valkey) will deploy a primary-replica cluster using Valkey Sentinel, the [Valkey Cluster Helm Chart](https://github.com/bitnami/charts/tree/main/bitnami/valkey-cluster) will deploy a Valkey Cluster with sharding.
 The main features of each chart are the following:
 
 | Valkey                                     | Valkey Cluster                                                   |
 |--------------------------------------------|------------------------------------------------------------------|
 | Supports multiple databases                | Supports only one database. Better if you have a big dataset     |
-| Single write point (single master)         | Multiple write points (multiple masters)                         |
+| Single write point (single primary)         | Multiple write points (multiple primarys)                         |
```
Comment (juan131) [inline bitnami/valkey-cluster/README.md:34]: I don't think `primarys` exist in English:
```suggestion
| Single write point (single primary)         | Multiple write points (multiple primary nodes)                       |
```

File: bitnami/valkey-cluster/README.md [hunk 2]
```diff
@@ -694,6 +694,12 @@ helm install my-release -f values.yaml oci://REGISTRY_NAME/REPOSITORY_NAME/valke
 
 Find more information about how to deal with common errors related to Bitnami's Helm charts in [this troubleshooting guide](https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues).
 
+## Upgrading
+
+### To 2.0.0
+
+This major updates all the references from `master/slave` to `primary/replica` to follow the upstream project strategy. Environment variables previously prefixed as `VALKEY_MASTER` or `VALKEY_SENTINEL_MASTER` use `VALKEY_PRIMARY` and `VALKEY_SENTINEL_PRIMARY` now.
```
Comment (juan131) [inline bitnami/valkey-cluster/README.md:701]: Missing upgrade instructions

File: bitnami/valkey-cluster/README.md [hunk 3]
```diff
@@ -81,20 +81,20 @@ To modify the application version used in this chart, specify a different versio
...
 
-The replicas will be read-only replicas of the masters. By default only one service is exposed (when not using the external access mode). You will connect your client to the exposed service, regardless you need to read or write. When a write operation arrives to a replica it will redirect the client to the proper master node. For example, using `valkey-cli` you will need to provide the `-c` flag for `valkey-cli` to follow the redirection automatically.
+The replicas will be read-only replicas of the primarys. By default only one service is exposed (when not using the external access mode). You will connect your client to the exposed service, regardless you need to read or write. When a write operation arrives to a replica it will redirect the client to the proper primary node. For example, using `valkey-cli` you will need to provide the `-c` flag for `valkey-cli` to follow the redirection automatically.
```
Comment (juan131) [inline bitnami/valkey-cluster/README.md:88]: Same here: `primarys` -> `primary nodes`
Review other references.

File: bitnami/valkey-cluster/templates/configmap.yaml [hunk 4]
```diff
@@ -214,7 +214,7 @@ data:
     # tls-port 6379
 
     # Configure a X.509 certificate and private key to use for authenticating the
-    # server to connected clients, masters or cluster peers.  These files should be
+    # server to connected clients, primarys or cluster peers.  These files should be
```
Comment (juan131) [inline bitnami/valkey-cluster/templates/configmap.yaml:217]: `primarys` references here too

File: bitnami/valkey-cluster/templates/update-cluster.yaml [hunk 5]
```diff
@@ -141,16 +141,16 @@ spec:
...
-                # when the index of the new node is greater than or equal to `slaveNodesEndPos`，and it is not a multiple of `$VALKEY_CLUSTER_REPLICAS + 1`, the added node is a replica that assigned to newly added master node
-                if (( $VALKEY_CLUSTER_REPLICAS >= 1 )) && (( (( $newNodeIndex < $slaveNodesEndPos )) || (( (( $newNodeIndex >= $slaveNodesEndPos )) && (( $newNodeIndex % (( $VALKEY_CLUSTER_REPLICAS + 1 )) )) )) )); then
-                  slave+=("--cluster-slave")
+                replica=()
+                # when the index of the new node is less than `replicaNodesEndPos`，the added node is a replica that assigned to original valkey primary node
+                # when the index of the new node is greater than or equal to `replicaNodesEndPos`，and it is not a multiple of `$VALKEY_CLUSTER_REPLICAS + 1`, the added node is a replica that assigned to newly added primary node
+                if (( $VALKEY_CLUSTER_REPLICAS >= 1 )) && (( (( $newNodeIndex < $replicaNodesEndPos )) || (( (( $newNodeIndex >= $replicaNodesEndPos )) && (( $newNodeIndex % (( $VALKEY_CLUSTER_REPLICAS + 1 )) )) )) )); then
+                  replica+=("--cluster-slave")
```
Comment (juan131) [inline bitnami/valkey-cluster/templates/update-cluster.yaml:148]: I guess here the "slave" term still applies, right?

**Review by juan131 — CHANGES_REQUESTED** [review]

File: bitnami/valkey-cluster/templates/update-cluster.yaml (hunk 5 above)
Comment (dgomezleon) [reply bitnami/valkey-cluster/templates/update-cluster.yaml:148]: Yes, I have not found a replacement for that flag, and that's working, so I prefer to keep it.

**Review by juan131 — APPROVED** [review]
LGTM

**Review by juan131 — APPROVED** [review]

**Pull request merged**